### PR TITLE
Cluster autoscaler friendly scheduling algorithm

### PR DESCRIPTION
**Pull request by mwielgus**
cc: @davidopp

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30992)

### CODE REVIEW

File: plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go [hunk 1]
```diff
@@ -95,6 +95,18 @@ func init() {
 	factory.RegisterFitPredicate("MatchNodeSelector", predicates.PodSelectorMatches)
 	// Optional, cluster-autoscaler friendly priority function - give used nodes higher priority.
 	factory.RegisterPriorityFunction("MostRequestedPriority", priorities.MostRequestedPriority, 1)
+	// Cluster autoscaler friendly scheduling algorigthm.
+	factory.RegisterAlgorithmProvider("ClusterAutoscalerProvider", defaultPredicates(),
```
Comment (wojtek-t) [inline plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go]: nit: can you please register it next the the other provider (at the beginning of this function)?

Comment (mwielgus) [inline plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go]: Done.

File: plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go [hunk 2]
```diff
@@ -54,6 +54,9 @@ func getMaxVols(defaultVal int) int {
 
 func init() {
 	factory.RegisterAlgorithmProvider(factory.DefaultProvider, defaultPredicates(), defaultPriorities())
+	// Cluster autoscaler friendly scheduling algorithm.
+	factory.RegisterAlgorithmProvider("ClusterAutoscalerProvider", defaultPredicates(),
```
Comment (wojtek-t) [inline plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go]: please make this a constant

Comment (mwielgus) [inline plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go]: Done.

**Comment by wojtek-t** [comment]
LGTM

**Pull request merged**